iamctl/pkg/utils/keywordReplacement.go

```diff
@@ -0,0 +1,237 @@
+
+	var localYaml interface{}
+	err := yaml.Unmarshal(localFileData, &localYaml)
+	if err != nil {
+		log.Println("Error: ", err)
+	}
+	// If the local file is empty or not available, return the exported data as it is.
+	if err == nil && localYaml != nil {
+		// Get keyword locations in local file.
+		keywordLocations := GetKeywordLocations(localYaml, []string{}, keywordMapping)
+		// Load exported app data as a yaml object.
+		var exportedYaml interface{}
+		err = yaml.Unmarshal(exportedData, &exportedYaml)
+		if err != nil {
+			log.Println("Error: ", err)
+		}
+
+		// Compare the fields with keywords in the exported file and the local file and modify the exported file.
+		exportedYaml = ModifyFieldsWithKeywords(exportedYaml, localYaml, keywordLocations, keywordMapping)
+
+		exportedData, err = yaml.Marshal(exportedYaml)
+		if err != nil {
+			panic(err)
+		}
+	}
+	return exportedData
+}
+
+func GetKeywordLocations(fileData interface{}, path []string, keywordMapping map[string]interface{}) []string {
+
+	var keys []string
+	switch v := fileData.(type) {
+	case map[interface{}]interface{}:
+		for k, val := range v {
+			newPath := append(path, fmt.Sprintf("%v", k))
+			keys = append(keys, GetKeywordLocations(val, newPath, keywordMapping)...)
+		}
+	case []interface{}:
+		for _, val := range v {
+			arrayElementPath, err := resolvePathWithIdentifiers(path[len(path)-1], val, appArrayIdentifiers)
+			if err != nil {
+				log.Println("Error: ", err)
+				break
+			} else {
+				newPath := append(path, arrayElementPath)
+				keys = append(keys, GetKeywordLocations(val, newPath, keywordMapping)...)
+			}
+		}
+	case string:
+		if ContainsKeywords(fileData.(string), keywordMapping) {
+			thisPath := strings.Join(path, ".")
+			keys = append(keys, thisPath)
+		}
+	}
+	return keys
+}
+
+func resolvePathWithIdentifiers(arrayName string, element interface{}, identifiers map[string]string) (string, error) {
+
+	elementMap, ok := element.(map[interface{}]interface{})
+	if !ok {
+		log.Printf("cannot convert %T to a map", element)
+	}
+	identifier := identifiers[arrayName]
+	// If an identifier is not defined for the array, use the default identifier "name".
+	if identifier == "" {
+		identifier = "name"
+	}
+	identifierValue, ok := elementMap[identifier]
+	if !ok {
+		return "", fmt.Errorf("identifier not found for the array %s", arrayName)
+	}
+	return fmt.Sprintf("[%s=%s]", identifier, identifierValue), nil
+}
+
+func ContainsKeywords(data string, keywordMapping map[string]interface{}) bool {
+
+	for keyword := range keywordMapping {
+		if strings.Contains(data, "{{"+keyword+"}}") {
+			return true
+		}
+	}
+	return false
+}
+
+func ModifyFieldsWithKeywords(exportedFileData interface{}, localFileData interface{},
+	keywordLocations []string, keywordMap map[string]interface{}) interface{} {
+
+	for _, location := range keywordLocations {
+
+		localValue := GetValue(localFileData, location)
+		localReplacedValue := ReplaceKeywords(localValue, keywordMap)
```

Comment:
what's the behaviour when the localValue is ""?

Reply:
If the localValue = "", ReplaceKeywords function will return an empty string "".